src/Components/EditInventory/EditInventory.js

```diff
@@ -0,0 +1,192 @@
+    const [nameError, setNameError] = useState(false);
+    const [descError, setDescError] = useState(false);
+    const [quantityError, setQuantityError] = useState(false);
+
+    useEffect (() => {
+        axios.get(`http://localhost:8080/inventory/${params.id}`).then(res => {
+            setItemDetails(res.data[0]);
+            updateTheStock(res.data[0]);
+        }).catch(err => {
+            console.log(err);
+        });
+        axios.get(`http://localhost:8080/warehouse/`).then(res => {
+            setWarehouses(res.data);   
+        });
+
+    }, [params]);
+
+    const updateTheStock = details => {
+        if(details.status === "In Stock" && details.quantity > 0) {
+            setInStock(true);
+        }
+        else {
+            setInStock(false);
+        }
+    }
+
+    const handleFormChange = e => {
+
+        if (e.target.name === 'warehouseName') {
+                const wh = warehouses.filter(wh => wh.name === e.target.value);
+                console.log(wh[0].id);
+                setItemDetails({...itemDetails, warehouseId: wh[0].id});
+        }
+        setItemDetails({...itemDetails, [e.target.name]: e.target.value});
+        if (e.target.name === 'status') {
+            if(e.target.value === "In Stock") {
+                setInStock(true);
+            }
+            else {
+                setInStock(false);
+            }
+        }
+    }
+
+    const handleSave = e => {
+        e.preventDefault();
+        console.log(e);
+        
+        e.target[0].classList.remove('error-state');
+        e.target[1].classList.remove('error-state');
+        e.target[5].classList.remove('error-state');
+
+        let emptyValue = false;
+        if (e.target[0].value === '') { 
+            e.target[0].classList.add('error-state');
+            setNameError(true);
+            emptyValue = true;
+        } else {
+            setNameError(false);
+        }
+        if (e.target[1].value === '') {
+            e.target[1].classList.add('error-state');
+            setDescError(true);
+            emptyValue = true;
+        } else {
+            setDescError(false);
+        }
+        if (e.target[5].value <= '0' && itemDetails.status === 'In Stock') {
+            e.target[5].classList.add('error-state');
+            setQuantityError(true);
+            emptyValue = true;
+        } else {
+            setQuantityError(false);
+        }
+
+        if (emptyValue) {
+            return false;
+        }
+        
+        console.log(itemDetails);
+        if (window.confirm('Are you sure you want to update this Inventory Item?')) {
+            
+            axios.put(`http://localhost:8080/inventory/${params.id}`, itemDetails).then(res => console.log(res)).catch(err => console.log(err));
+            history.push(`/inventory/${params.id}`);
+        };
+    }
+
+    const handleCancel = () => {
+        if(window.confirm('Are yo sure you want to cancel and go back to the previous page?')) {
```

Comment:
I don't remember seeing a requirement for a window confirmation screen, but I do like it. However, the text here is a bit off if we've started editing the inventory item from the main inventory list, as our last route was simply just `/inventory` instead of `/inventory/:id`